### PR TITLE
chore(docs): Rework CI docs

### PR DESCRIPTION
**Pull request by Skn0tt**
This PR moves around some of our CI docs. It moves the GitHub actions docs from `ci-intro.md` to `ci.md`, reduces `ci-intro.md` to be an introduction, adds a mention of Sharding to the best practices, and adds a section on `--only-changed` called "Fail-Fast". Each of those changes is a separate commit, to make this a little easier to review. If we find any of those to commits to be contentious, i'll pull them out into individual PRs.

While rolling this to playwright.dev, we'll also make the following changes to its sidebar:
- move the `ci.md` document from the "Integrations" section to the "Playwright Test" section
- make "Best Practices" the last item of the "Getting Started" section

### CODE REVIEW

File: docs/src/ci.md [hunk 1]
```diff
@@ -59,14 +59,397 @@ export default defineConfig({
 });
 ```
 
-
 ## CI configurations
 
+### GitHub Actions
+
 The [Command line tools](./browsers#install-system-dependencies) can be used to install all operating system dependencies on GitHub Actions.
```
Comment (mxschmitt) [inline docs/src/ci.md:66]: that should be not part of `### GitHub Actions` I think and moved one layer above.

File: docs/src/ci-intro.md [hunk 2]
```diff
@@ -57,25 +51,34 @@ jobs:
...
-        node-version: 18
+        node-version: lts/*
     - name: Install dependencies
       run: npm ci
     - name: Install Playwright Browsers
       run: npx playwright install --with-deps
     - name: Run Playwright tests
       run: npx playwright test
     - uses: actions/upload-artifact@v4
-      if: ${{ !cancelled() }}
```
Comment (mxschmitt) [inline docs/src/ci-intro.md:68]: revert that?

File: docs/src/ci.md (hunk 1 above)
Comment (Skn0tt) [reply docs/src/ci.md:66]: The sentence contains "GitHub Actions". If it applies to more CI providers, then we'll need to reword it.

File: docs/src/ci-intro.md (hunk 2 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md:68]: I've copied this over from the "on push/pull_request" recipe in `ci.md`. Let's try to use the same snippet between the two - which one do you prefer?

File: docs/src/ci.md (hunk 1 above)
Comment (mxschmitt) [reply docs/src/ci.md:66]: I'd reword, e.g. on your CI provider.

File: docs/src/ci-intro.md (hunk 2 above)
Comment (mxschmitt) [reply docs/src/ci-intro.md:68]: `if: ${{ !cancelled() }}` is what we changed it to based on https://github.com/microsoft/playwright/pull/30102

Comment (Skn0tt) [reply docs/src/ci-intro.md:68]: hmm, then that missed one of the recipes. I'll update it!

File: docs/src/ci.md [hunk 3]
```diff
@@ -59,14 +59,397 @@ export default defineConfig({
...
```
Comment (Skn0tt) [inline docs/src/ci.md:67]:
```suggestion

The [Command line tools](./browsers#install-system-dependencies) can be used to install all operating system dependencies in CI.

### GitHub Actions

```

Comment (Skn0tt) [reply docs/src/ci.md:67]: @mxschmitt how do you like this?

File: docs/src/ci-intro.md [hunk 4]
```diff
@@ -158,268 +161,14 @@ jobs:
...
-          dotnet-version: 8.0.x
-      - run: dotnet build
-      - name: Run your tests
-        run: dotnet test
-        env:
-          HOME: /root
-```
-
-### On deployment
+If this doesn't make sense to you, give GitHub's [Understanding GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) guide a read.
```
Comment (debs-obrien) [inline docs/src/ci-intro.md:164]: i dont like the wording here. it reads as if you are not sure if you explained it well. 
Best to put something like To learn more about GitHub Actions or something along those lines

File: docs/src/ci-intro.md (hunk 2 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md]: done in https://github.com/microsoft/playwright/pull/31988/commits/acbcbf1b472be396429dd8b49d1058eb2b235745

Comment (Skn0tt) [reply docs/src/ci-intro.md]: I think I got confused here because I copied over from `create-playwright`. Looks like we missed that when updating, here's a PR to also use `!cancelled()` there: https://github.com/microsoft/create-playwright/pull/128

File: docs/src/ci-intro.md [hunk 5]
```diff
@@ -57,25 +51,34 @@ jobs:
...
-      if: ${{ !cancelled() }}
+      if: always()
       with:
         name: playwright-report
         path: playwright-report/
         retention-days: 30
 ```
 
-### On push/pull_request
+If this doesn't make sense to you, give GitHub's [Understanding GitHub Actions](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions) guide a read. Looking at the list of steps in `jobs.test.steps`, you can see that the workflow performs these steps:
```
Comment (debs-obrien) [inline docs/src/ci-intro.md:69]: same comment as below regarding wording if this doesnt make sense

File: docs/src/ci-intro.md [hunk 6]
```diff
@@ -25,22 +23,18 @@ When [installing Playwright](./intro.md) using the [VS Code extension](./getting
 
 Playwright tests can be ran on any CI provider. In this section we will cover running tests on GitHub using GitHub actions. If you would like to see how to configure other CI providers check out our detailed doc on Continuous Integration.
 
-To add a [GitHub Actions](https://docs.github.com/en/actions) file first create `.github/workflows` folder and inside it add a `playwright.yml` file containing the example code below so that your tests will run on each push and pull request for the main/master branch.
-
 #### You will learn
 * langs: python, java, csharp
```
Comment (debs-obrien) [inline docs/src/ci-intro.md:27]: just double check this makes sense for all langs except js or it should also have js?

File: docs/src/ci-intro.md (hunk 4 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md:164]:
```suggestion
To learn more about this, see ["Understanding GitHub Actions"](https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions).
```

How do you like this?

File: docs/src/ci-intro.md (hunk 6 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md:27]: Yes this is intended. There's a separate "You will learn" list for JS that has different contents in https://github.com/microsoft/playwright/pull/31988/files#diff-49b43d48c93c3dc274299a0604e3a7c8ea1ed9891a230f3125e9992867d2ed3fL13-L14.

File: docs/src/ci-intro.md (hunk 4 above)
Comment (debs-obrien) [reply docs/src/ci-intro.md:164]: Perfect. Now just to change it in intro doc too

File: docs/src/ci-intro.md (hunk 5 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md]: done in https://github.com/microsoft/playwright/pull/31988/commits/34ccc7a1e17b7d114bd3018344ef7ea3864ce280

File: docs/src/best-practices-js.md [hunk 7]
```diff
@@ -261,7 +261,7 @@ npx playwright --version
 
 Setup CI/CD and run your tests frequently. The more often you run your tests the better. Ideally you should run your tests on each commit and pull request. Playwright comes with a [GitHub actions workflow](/ci-intro.md) so that tests will run on CI for you with no setup required. Playwright can also be setup on the [CI environment](/ci.md) of your choice.
 
-Use Linux when running your tests on CI as it is cheaper. Developers can use whatever environment when running locally but use linux on CI.
+Use Linux when running your tests on CI as it is cheaper. Developers can use whatever environment when running locally but use linux on CI. Set up [Sharding](./test-sharding.md) to make CI faster.
```
Comment (yury-s) [inline docs/src/best-practices-js.md:264]:
```suggestion
Use Linux when running your tests on CI as it is cheaper. Developers can use whatever environment when running locally but use linux on CI. Consider setting up [Sharding](./test-sharding.md) to make CI faster.
```
Not everybody needs sharding and for some projects it may be harmful.

File: docs/src/ci-intro.md [hunk 8]
```diff
@@ -126,7 +131,7 @@ jobs:
       with:
         distribution: 'temurin'
         java-version: '17'
-    - name: Build & Install
+    - name: Ensure browsers are installed
```
Comment (yury-s) [inline docs/src/ci-intro.md:134]: Revert this? The step actually triggers build of the project (mvn compile) and then installs the artifacts into local maven cache. It does *not* install browsers, they will be downloaded on first playwright run.

File: docs/src/ci-intro.md [hunk 9]
```diff
@@ -126,7 +131,7 @@ jobs:
...
       run: mvn -B install -D skipTests --no-transfer-progress
     - name: Install Playwright
```
Comment (yury-s) [inline docs/src/ci-intro.md:136]:
```suggestion
    - name: Ensure browsers are installed
```

File: docs/src/ci-intro.md [hunk 10]
```diff
@@ -158,268 +163,14 @@ jobs:
...
-      run: npx playwright test
-    - uses: actions/upload-artifact@v4
-      if: ${{ !cancelled() }}
-      with:
-        name: playwright-report
-        path: playwright-report/
-        retention-days: 30
-```
+1. Clone your repository
+2. Install language dependencies
```
Comment (yury-s) [inline docs/src/ci-intro.md:171]: 3. Build project ?

File: docs/src/ci.md (hunk 1 above)
Comment (yury-s) [inline docs/src/ci.md:66]: I don't understand who this sentence is addressed to under `GitHub Actions` title. Previously the user was redirected to our comprehensive getting started guide which is where they should start. If they know that stuff, they likely now how to install browsers and deps already. I'd revert `Check out our [GitHub Actions](ci-intro.md) guide` here or maybe just drop this paragraph.

Comment (Skn0tt) [reply docs/src/ci.md:66]: This PR moves the comprehensive github actions guide into here, so `ci-intro.md` is just an intro. I agree that the paragraph reads weird though, max was saying the same in https://github.com/microsoft/playwright/pull/31988#discussion_r1701947754. I'll move the paragraph above the heading and reword.

File: docs/src/ci-intro.md (hunk 10 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md:172]: yes! done in https://github.com/microsoft/playwright/pull/31988/commits/59ae200e0cda728b51bc2a464b8fe7a14ef16ebe, and i've also updated the step names to be the same across the different CI snippets.

File: docs/src/ci-intro.md (hunk 9 above)
Comment (Skn0tt) [reply docs/src/ci-intro.md]: good point, not sure what went wrong here. I've copied over the snippet from `ci.md` to keep them in sync. https://github.com/microsoft/playwright/pull/31988/commits/0be85a9ae65c679d35fe2e93f0b3576f758bd7b0

**Review by yury-s — APPROVED** [review]
